react/features/base/tracks/actions.js

```diff
@@ -103,12 +150,21 @@ export function createLocalTracksA(options = {}) {
  * @returns {Function}
  */
 export function destroyLocalTracks() {
-    return (dispatch, getState) =>
-        dispatch(
-            _disposeAndRemoveTracks(
+    return (dispatch, getState) => {
+        // First wait until any getUserMedia in progress is settled and then get
+        // rid of all local tracks.
+        Promise
+            .all(
                 getState()['features/base/tracks']
                     .filter(t => t.local)
-                    .map(t => t.jitsiTrack)));
+                    .map(t => t.gumProcess && t.gumProcess.cancel()))
```

Comment:
@paweldomas, I'm not sure about `Promise.all(all gumProcesses)` because I don't think we care whether they resolve (i.e. succeed) but I'd rather think that we care that they settle (i.e. even if they are rejected). With `Promise.all` we will not get to `_disposeAndRemoveTracks` if at least one `gumProcess` is rejected. Could you please take a look?